toponn-utils/tensorflow/model.py

```diff
@@ -0,0 +1,131 @@
+import tensorflow as tf
+
+
+class Model:
+    def __init__(self, config, shapes):
+        self._config = config
+        self._shapes = shapes
+
+    def accuracy(self, prefix, predictions, labels):
+        correct = tf.equal(predictions, labels)
+
+        # Mask inactive timesteps
+        correct = tf.multiply(self.mask, tf.cast(correct, tf.float32))
+
+        # Compensate for inactive time steps.
+        correct = tf.reshape(correct, [-1])
+        correct = tf.truediv(correct, tf.reduce_mean(self.mask))
+
+        return tf.reduce_mean(correct, name="%s_accuracy" % prefix)
+
+    def affine_transform(self, prefix, x, n_outputs):
```

Comment:
Is there a difference in performance to tf.layers.dense(n_outputs, use_bias=True, activation=None)

Reply:
Looking at the implementation of the dense layer, it should boil down to the same. But since it's better to use canonical TF functions, I'll update this to use `dense`. Thanks for pointing this our!